src/components/tl/nccl/tl_nccl_team.c

```diff
@@ -22,9 +30,10 @@ UCC_CLASS_INIT_FUNC(ucc_tl_nccl_team_t, ucc_base_context_t *tl_context,
     UCC_CLASS_CALL_SUPER_INIT(ucc_tl_team_t, &ctx->super, params);
 
     size = UCC_TL_TEAM_SIZE(self);
-    self->comm_state = UCC_OK;
-    self->unique_id  = ucc_malloc(sizeof(ncclUniqueId) * (size + 1),
-                                  "tl_nccl_unique_id");
+    self->comm_state    = UCC_OK;
+    self->nccl_nb_state = NCCL_NB_UNUSED;
```

Comment:
because id comm_state is UCC_OK we will still do ncclCommFinalize() everytime we come back from in progress, in which case we want to skip and go straight to checking finalize state with ncclCommGetAsyncError()

Reply:
changed to use comm state with UCC_INPROGRESS